tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -782,56 +847,35 @@ def generator_py_func(iterator_id):
         """A `py_func` that will be called to invoke the iterator."""
         # `next()` raises `StopIteration` when there are no more
         # elements remaining to be generated.
-        values = next(generator_state.get_iterator(iterator_id))
+        values = next(generator_state.get_iterator(iterator_id.numpy()))
```

Comment:
Why the extra call to `numpy()` here? This code could be executed in graph-mode.